File: backend/app/schemas/instance.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, Dict
from .target import Target

class InstanceBase(BaseModel):
    name: str
    target_id: int
    ports: Optional[Dict[str, str]] = None  # 例如: {"80/tcp": "8080"}
    environment: Optional[Dict[str, str]] = None  # 例如: {"MYSQL_ROOT_PASSWORD": "123456"}

class InstanceCreate(InstanceBase):
    pass

class InstanceUpdate(BaseModel):
    name: Optional[str] = None
    status: Optional[str] = None

class Instance(InstanceBase):
    id: int
    container_id: Optional[str] = None
    status: str
    created_at: datetime
    created_by_id: int
    target: Target

    class Config:
        orm_mode = True 

File: frontend/src/types/instance.ts
export interface Instance {
  id: number
  name: string
  status: 'running' | 'stopped' | 'error'
  container_id: string
  target_id: number
  port_mappings: string
  created_at: string
  created_by_id: number
  target?: {
    id: number
    name: string
    image?: {
      registry_path: string
    }
  }
}

export interface InstanceCreate {
  name: string
  target_id: number
  port_mappings?: string
}

export interface InstanceUpdate {
  name?: string
  status?: 'running' | 'stopped'
}

export interface PortMapping {
  container_port: number
  host_port: number
  protocol: 'tcp' | 'udp'
} 